src/state/charts/sagas/GetSupersetGuestToken.js

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { put, select, takeLatest } from 'redux-saga/effects';
+import { STATUSES } from '../../../services/config/StatusConstants';
+import { SupersetService } from '../../../services/superset/SupersetService';
+import { CHART_ACTIONS_KEY } from '../constants';
+import { setSupersetGuestToken } from '../reducers';
+
+const getSupersetChartsConfig = (state) => state?.workspace?.current?.data?.webApp?.options?.charts;
+const getOrganizationId = (state) => state?.organization?.current?.data?.id;
+const getWorkspaceId = (state) => state?.workspace?.current?.data?.id;
+
+function extractDashboardIds(chartsConfig) {
+  const dashboardIds = new Set();
+
+  if (chartsConfig?.scenarioView) {
+    const scenarioView = chartsConfig.scenarioView;
+
+    if (Array.isArray(scenarioView)) {
+      scenarioView.forEach((dashboard) => {
+        if (dashboard?.dashboardId) {
+          dashboardIds.add(dashboard.dashboardId);
+        }
+      });
+    } else if (typeof scenarioView === 'object') {
+      if (scenarioView.default) {
+        dashboardIds.add(scenarioView.default);
+      }
+      if (scenarioView.overrides) {
+        Object.values(scenarioView.overrides).forEach((id) => {
+          if (id) dashboardIds.add(id);
+        });
+      }
+    }
+  }
+
+  const dashboardView = chartsConfig?.dashboardView || chartsConfig?.dashboardsView;
+  if (dashboardView && Array.isArray(dashboardView)) {
+    dashboardView.forEach((dashboard) => {
+      if (dashboard?.id) {
+        dashboardIds.add(dashboard.id);
+      } else if (dashboard?.dashboardId) {
+        dashboardIds.add(dashboard.dashboardId);
+      }
+    });
+  }
+
+  if (chartsConfig?.dashboards && Array.isArray(chartsConfig.dashboards)) {
```

Comment:
Same as above.
You can directly write :` if (Array.isArray(chartsConfig?.dashboards))`